reana_workflow_controller/rest.py

```diff
@@ -1417,6 +1428,36 @@ def _start_workflow(workflow, parameters):
         raise REANAWorkflowControllerError(message)
 
 
+def _stop_workflow(workflow):
+    """Stop a given workflow."""
+    if workflow.status == WorkflowStatus.running:
+        workflow.run_stopped_at = datetime.now()
+        workflow.status = WorkflowStatus.stopped
+        current_db_sessions = Session.object_session(workflow)
+        current_db_sessions.add(workflow)
+        current_db_sessions.commit()
+        kwargs = {
+            'workflow_uuid': str(workflow.id_),
+            'job_list':
+            workflow.job_progress.get('running', {}).get('job_ids', []),
+        }
+        if workflow.type_ in ['serial', 'yadage', 'cwl']:
+            stop_workflow.apply_async(
+                kwargs=kwargs,
+                queue=WORKFLOW_QUEUES[workflow.type_],
+                task_id='delete-{}'.format(str(workflow.id_))
+            )
+        else:
+            raise NotImplementedError(
+                'Workflow type {} does not support '
+                'stop yet.'.format(workflow.type_))
+    else:
+        message = \
+            ("Workflow {id_} could not be stopped. Only running workflows"
```

Comment:
reana-client has the same first line in the error message https://github.com/reanahub/reana-client/pull/205/files#diff-4f27f08c3f46f0b755c1d5ac1a1e2dcaR445 and stopping a not running workflow shows:
```
Workflow could not be stopped: 
Workflow 7044131f-bc73-455d-94aa-a56c83cf8677 could not be stopped. Only running workflows can be stopped.
```
This message could be sth like "Workflow {id_} is not running."